modules/src/ics03_connection/connection.rs

```diff
@@ -66,6 +68,21 @@ impl ConnectionEnd {
         })
     }
 
+    pub fn test_channel_new(
```

Comment:
This method should be moved under the `tests` module in `ics04_channel/handler/chan_open_init.rs`.

Reply:
Reason for that is that we don't want to include test-only code in the public API of the project.
As an additional benefit, since the `tests` module in `ics04_channel/handler/chan_open_init.rs` is defined as 

```rust
#[cfg(test)]
mod tests {
  ...
}
```

it will only be compiled when building the tests, eg. via `cargo test`.

Reply:
yeah, this is indeed ugly. It appeared when I gave up on style to have at least one test. Compiling failed me because fields have the same name in different structures. :) In this case State is a field name in both in connection and Channel.  I need help understanding how I can create a connection in channel without rust complaining that is expects State to belong to either connection or channel.